File: app/src/main/java/black/bracken/shotsorter/AndroidBootReceiver.java
package black.bracken.shotsorter;

import android.annotation.SuppressLint;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import black.bracken.shotsorter.constant.PreferencesKeys;
import black.bracken.shotsorter.service.SortService;

/**
 * @author BlackBracken
 */
public final class AndroidBootReceiver extends BroadcastReceiver {

    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    @Override
    public void onReceive(Context context, Intent intent) {
        if (context.getSharedPreferences(PreferencesKeys.ROOT, Context.MODE_PRIVATE).getBoolean(PreferencesKeys.RUN_ON_STARTUP, false)) {
            SortService.start(context);
        }
    }

}

File: app/src/main/java/black/bracken/shotsorter/AndroidBootReceiver.kt
package black.bracken.shotsorter

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import black.bracken.shotsorter.data.repository.singleSettingRepositoryModule
import black.bracken.shotsorter.domain.repository.SettingRepository
import black.bracken.shotsorter.service.SortService
import org.kodein.di.Kodein
import org.kodein.di.KodeinAware
import org.kodein.di.generic.instance

/**
 * @author BlackBracken
 */
class AndroidBootReceiver : BroadcastReceiver(), KodeinAware {

    override val kodein = Kodein {
        import(singleSettingRepositoryModule)
    }

    private val settingRepository: SettingRepository by instance()

    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    override fun onReceive(context: Context, intent: Intent) {
        if (settingRepository.shouldRunOnStartup) {
            SortService.start(context)
        }
    }

}
